staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/promise/promise.go

```diff
@@ -39,20 +40,32 @@ type promise struct {
 var _ WriteOnce = &promise{}
 
 // NewWriteOnce makes a new promise.LockingWriteOnce
-func NewWriteOnce(lock sync.Locker, activeCounter counter.GoRoutineCounter) WriteOnce {
+func NewWriteOnce(activeCounter counter.GoRoutineCounter) WriteOnce {
 	return &promise{
-		cond:          *sync.NewCond(lock),
+		doneCh:       make(chan struct{}),
 		activeCounter: activeCounter,
 	}
 }
 
-func (p *promise) Get() interface{} {
+// FIXME:
+var timeoutErr = fmt.Errorf("timeout")
+
+func (p *promise) Get(lock sync.Locker, stopCh <-chan struct{}) (interface{}, error) {
 	if !p.isSet {
 		p.waitingCount++
 		p.activeCounter.Add(-1)
-		p.cond.Wait()
+
+		lock.Unlock()
```

Comment:
I don't think the locking pattern is what we really want - but that was supposed to illustrate the feasibility.